src/borg/cache.py

```diff
@@ -564,6 +564,9 @@ def create_master_idx(chunk_idx):
             # deallocates old hashindex, creates empty hashindex:
             chunk_idx.clear()
             cleanup_outdated(cached_ids - archive_ids)
+            # Explicitly set the initial hash table capacity to avoid performance issues
+            # due to hash table "resonance".
+            master_index_capacity = int(len(self.repository) / ChunkIndex.MAX_LOAD_FACTOR)
```

Comment:
hmm, yeah, as the repo contains all the chunks referenced by archives, the computation is ok.

good idea about giving usable capacity, separate PR just for that.